R/DataMart.R

```diff
@@ -127,44 +153,56 @@ QLoadData <- function(filename)
         msg <- ifelse(req$status_code == 404, "File not found.", "Could not connect to Data Mart.")
         stop(msg)
     }
-        
-    obj <- readRDS(tmpfile)
-    invisible(file.remove(tmpfile))
-    return (obj)
-    
+    if (file.exists(tmpfile)) 
+    {
+        obj <- readRDS(tmpfile) 
+        file.remove(tmpfile)
+        return (obj)
+    }
+    stop("Could not read from file.")
 }
 
-#' Saves an object into an rds file blob
+#' Save an object
+#' 
+#' Saves an object to the Data Mart as an *.rds file of the given name.
+#' If the name is given an extension, this must be '.rds'. If there is no
+#' extension specified, this defaults to '.rds'.
+#' 
 #' @param object object. The object to be uploaded
 #' @param filename character string. Name of the file to be written to
+#' 
+#' @return 
+#' 
 #' @importFrom httr POST add_headers upload_file
+#' @importFrom tools file_ext
+#' 
 #' @export 
 QSaveData <- function(object, filename)
 {
     if (file_ext(filename) == "") 
         filename <- append(filename, ".rds")
     
     if (file_ext(filename) != "rds")
-        stop("File-type must be .rds")
+        stop("File must be of type *.rds")
     
     tmpfile <- tempfile()
     saveRDS(object, tmpfile)
+    on.exit(if(file.exists(tmpfile)) file.remove(tmpfile))
     
-    res <- try(POST(paste0(url, "?filename=", filename), 
+    companySecret <- ifelse(exists("companySecret"), companySecret, "")
```

Comment:
`ifelse()` is not great for a few reasons. you can replace this with `companySecret <- get0("companySecret", ifnotfound = "")`